test/testAnnoutil.pl

```diff
@@ -0,0 +1,2223 @@
+        "rast_genome_assembly returns a VALID ref: $rast_ret->{output_genome_ref}");
+
+    $parms = {
+        "object_ref" => $obj_asmb,
+        "output_genome_name" => "rasted_assembly",
+        "output_workspace" => $ws
+    };
+    lives_ok {
+        $rast_ret = $rast_impl->rast_genome_assembly($parms);
+    } 'Impl rast_genome call returns without annotation due to local assembly did not run.';
+    ok (!defined($rast_ret ->{output_genome_ref}),
+        "due to local annotation on assembly with empty features, no rast was run.");
+};
+
+
+## testing _get_bulk_rast_parameters using obj ids from prod ONLY
+subtest '_get_bulk_rast_parameters' => sub {
+    my $parms = {
+          'input_genomes' => [
+                               '55141/242/1',
+                               '55141/212/1'
+                             ],
+          'create_report' => 0,
+          'output_workspace' => 'test_RAST_SDK_1592592098000',
+          'output_GenomeSet_name' => 'out_genomeSet',
+          'input_assemblies' => [
+                                  '55141/266/3',
+                                  '55141/243/1'
+                                ],
+          'input_text' => '55141/266/3;55141/212/1;63171/394/1'
+    };
+    my @refs = split(';', $parms->{input_text});
+    push(@refs, @{$parms->{input_assemblies}});
+    push(@refs, @{$parms->{input_genomes}});
+
+    my $params;
+    lives_ok {
+        $params = $annoutil->_get_bulk_rast_parameters($parms);
+    } "anno_utils _get_bulk_rast_parameters call returns normally.";
+
+    for my $ref (@refs) {
+        ok ($annoutil->_value_in_array($ref, $params), "$ref is included in the parameters\n");
+    }
+
+    # testing invalid object name/id inputs
+    $parms->{input_text} = '55141/266/3;55141/212/1;63171/394/1;abc/3/2;333/22/a';
+    @refs = split(';', $parms->{input_text});
+
+    lives_ok {
+        $params = $annoutil->_get_bulk_rast_parameters($parms);
+    } "anno_utils _get_bulk_rast_parameters call returns normally.";
+    for my $ref (@refs) {
+        if ($ref ne 'abc/3/2' && $ref ne '333/22/a') {
+            ok ($annoutil->_value_in_array($ref, $params), "$ref is included in the parameters\n");
+		}
+        else {
+            ok ($annoutil->_value_in_array($ref, $params) == 0, "invalid reference $ref is excluded from the parameters\n");
+		}
+    }
+};
+
+
+## testing Impl_rast_genomes_assemblies using obj ids from prod ONLY
+subtest 'rast_genomes_assemblies' => sub {
+    my $params = {
+        "output_GenomeSet_name" => "out_genomeSet"
+    };
+
+    lives_ok {
+        $params->{output_workspace} = get_ws_name();
+        $params->{input_genomes} = [$obj_Ecoli]; # array of prod objects
+        $params->{input_assemblies} = [$obj_asmb]; # array of prod objects
+        $params->{input_text} = '';
+        my $ret_ann6 = $rast_impl->rast_genomes_assemblies($params);
+    } "rast_impl rast_genomes_assemblies call on 1N1 returns normally.";
```

Comment:
Instead of checking that something doesn't die, test that the results are what you expect them to be.